pkg/reconciler/reconcile.go

```diff
@@ -9,7 +9,22 @@ import (
 	"go.uber.org/zap"
 )
 
-func Reconcile(logger *zap.Logger, client *db.Database, req forman.Request) forman.Result {
+type FleetReconciler struct {
+	// Data required for reconcile phases
+}
+
+func (r *FleetReconciler) Reconcile(ctx context.Context) (forman.Result, error) {
+	// Reconciles the desired state
+	// Example for an installation flow:
+	// 	1. Ensures the infra via AWS Cloudformation
+	// 	2. Ensures ROKS Cluster installation on existing infra
+	// 	3. Ensures ManagedFusionAgent Operator on ROKS Cluster
+	// 	4. Ensures presence of config for MFAO monitoring stack
+	// 	5. Ensures Data Foundation Config is present on ROKS Cluster
+	return forman.Result{}, nil
```

Comment:
Can we at least save the context on the reconciler?

Reply:
- I might need to wrap zap logger with logr and pass that interface, can I go head or a do you want simple `fmt` for now?

Reply:
@Madhu-1 already checked that, is seems that the logr interface is part of the operator framework which I think we should not introduce a dependency one. Can you just pass on the actual zap logger?

Reply:
here is the PR for it, https://github.com/red-hat-storage/managed-fusion-fleet-reconciler/pull/9, I looked at it with a fresh mind and found a easy solution 🗡️

Reply:
Great! I have looked at that PR and have a single [request](https://github.com/red-hat-storage/managed-fusion-fleet-reconciler/pull/9#issuecomment-1651422894)